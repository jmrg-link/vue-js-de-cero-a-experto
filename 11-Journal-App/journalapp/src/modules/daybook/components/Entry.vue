<template>
  <div class="entry-container mb-3 pointer p-2"
    @click="$router.push( { name: 'entry' , params: { id:10 } } )"
  >
    <!--Title-->
    <div class="entry-title d-flex">
      <span class="text-success fs-5 fw-bold">15</span>
      <span class="mx-1 fs-5">Julio</span>
      <span class="mx-2 fs-5 fw-light">2021, Jueves</span>
    </div>
    <div class="entry-description">
      Lorem ipsum dolor sit amet, consectetur adipisicing elit. Aperiam blanditiis cupiditate debitis saepe sapiente tempora! Eaque ex expedita iusto quos. Animi laborum quam quisquam? Ad asperiores molestiae ratione reprehenderit voluptate?
    </div>
  </div>

</template>

<script>
export default {

}
</script>
<style lang="scss" scoped>
  .entry-container{
    border-bottom: 1px solid #2c3e50;
    transition: 0.2s all ease-in;
    &:hover{
      background-color: lighten($color:grey, $amount: 45) ;
      transition: 0.2s all ease-in;
    }
    .entry-description{
      font-size: 12px;
    }
  }

</style>