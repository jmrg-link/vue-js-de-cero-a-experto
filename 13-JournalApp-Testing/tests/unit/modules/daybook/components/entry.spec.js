import {shallowMount} from "@vue/test-utils";
import Entry from "@/modules/daybook/components/Entry.vue";
import {journalState} from "../../../mock-data/test-journal-state";

describe("Pruebas en Entry Component", () => {
    const mockRouter = {
        push: jest.fn(),
    };
    const wrapper = shallowMount(Entry, {
        props: {
            entry: journalState.entries[0]
        },
        global: {
            mocks: {
                $router: mockRouter,
            },
        },
    });

    test("debe de hacer match con el snapshot", () => {
        expect(wrapper.html()).toMatchSnapshot();
    });

    test("debe de redireccionar al hacer click en el entry-container", () => {
        const entryContainer = wrapper.find('.entry-container') // genero una constante donde llamare al wrapper.find para buscar una clase
        entryContainer.trigger('click') // llamamos a la clase del wrapper y generamos un trigger que dispara el evento click
        expect(mockRouter.push).toHaveBeenCalledWith({ /* espera que : mockRouter.push de (jest.fn) halla sido llamado con algo {toHaveBeenCalledWith}  */
            name: 'entry',
            params: {
                id: journalState.entries[0].id
            }
        })
    })
    test("pruebas en las props computed", () => {
        // console.log(wrapper.vm) puede mostrar las props computed
        expect(wrapper.vm.day).toBe(9) //
        expect(wrapper.vm.months).toBe("Diciembre") //
        expect(wrapper.vm.yearDay).toBe("2021, Jueves") //
    });
});
