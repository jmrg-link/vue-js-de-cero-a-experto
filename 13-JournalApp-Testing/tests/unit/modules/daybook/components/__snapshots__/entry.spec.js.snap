// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en Entry Component debe de hacer match con el snapshot 1`] = `
<div class="entry-container mb-3 pointer p-2">
  <!--Title-->
  <div class="entry-title d-flex"><span class="text-success fs-5 fw-bold">9</span><span class="mx-1 fs-5">Diciembre</span><span class="mx-2 fs-5 fw-light">2021, Jueves</span></div>
  <div class="entry-description">nueva entrada...</div>
</div>
`;
